runtime/core/decoder/ctc_prefix_beam_search.h

```diff
@@ -33,12 +36,41 @@ struct PrefixScore {
   std::vector<int> times_s;           // times of viterbi blank path
   std::vector<int> times_ns;          // times of viterbi none blank path
 
-  PrefixScore() = default;
   float score() const { return LogAdd(s, ns); }
   float viterbi_score() const { return v_s > v_ns ? v_s : v_ns; }
   const std::vector<int>& times() const {
     return v_s > v_ns ? times_s : times_ns;
   }
+
+  bool has_context = false;
+  ContextState context_state;
+  float context_score = 0;
+  std::vector<std::pair<int, int>> boundaries;
```

Comment:
it is more clear if we use two vector: start_boundaries and end_boundaries. 
std::vector<int> start_boundaries;
std::vector<int> end_boundaries;
Both the logic and the post processing will be more clear.